src/main/java/ch/njol/skript/structures/StructCommand.java

```diff
@@ -0,0 +1,360 @@
+					}
+				})
+				.addEntryData(new KeyValueStructureEntryData<List<String>>("aliases", new ArrayList<>()) {
+					private final Pattern pattern = Pattern.compile("\\s*,\\s*/?");
+
+					@Override
+					protected List<String> getValue(String value) {
+						List<String> aliases = new ArrayList<>(Arrays.asList(pattern.split(value)));
+						if (aliases.get(0).startsWith("/")) {
+							aliases.set(0, aliases.get(0).substring(1));
+						} else if (aliases.get(0).isEmpty()) {
+							aliases = new ArrayList<>(0);
+						}
+						return aliases;
+					}
+				})
+				.addEntryData(new KeyValueStructureEntryData<Integer>("executable by", ScriptCommand.CONSOLE | ScriptCommand.PLAYERS) {
+					private final Pattern pattern = Pattern.compile("\\s*,\\s*|\\s+(and|or)\\s+");
+
+					@Override
+					@Nullable
+					protected Integer getValue(String value) {
+						int executableBy = 0;
+						for (String b : pattern.split(value)) {
+							if (b.equalsIgnoreCase("console") || b.equalsIgnoreCase("the console")) {
+								executableBy |= ScriptCommand.CONSOLE;
+							} else if (b.equalsIgnoreCase("players") || b.equalsIgnoreCase("player")) {
+								executableBy |= ScriptCommand.PLAYERS;
+							} else {
+								return null;
+							}
+						}
+						return executableBy;
+					}
+				})
+				.addEntryData(new LiteralStructureEntryData<>("cooldown", true, Timespan.class))
+				.addEntryData(new KeyValueStructureEntryData<VariableString>("cooldown message", true) {
+					@Override
+					@Nullable
+					protected VariableString getValue(String value) {
+						return VariableString.newInstance(value.replace("\"", "\"\""));
+					}
+				})
+				.addEntry("cooldown bypass", true)
+				.addEntryData(new KeyValueStructureEntryData<VariableString>("cooldown storage", true) {
+					@Override
+					@Nullable
+					protected VariableString getValue(String value) {
+						return VariableString.newInstance(value, StringMode.VARIABLE_NAME);
+					}
+				})
+				.addSection("trigger", false)
+				.build(),
+			"command <.+>"
+		);
+	}
+
+	@Nullable
+	private ScriptCommand scriptCommand;
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private EntryContainer entryContainer;
+
+	@Override
+	public boolean init(Literal<?>[] args, int matchedPattern, ParseResult parseResult, EntryContainer entryContainer) {
+		this.entryContainer = entryContainer;
+		return true;
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public void load() {
+		getParser().setCurrentEvent("command", CommandEvent.class);
+
+		String fullCommand = entryContainer.getSource().getKey();
+		assert fullCommand != null;
+		fullCommand = ScriptLoader.replaceOptions(fullCommand);
+
+		int level = 0;
+		for (int i = 0; i < fullCommand.length(); i++) {
+			if (fullCommand.charAt(i) == '[') {
+				level++;
+			} else if (fullCommand.charAt(i) == ']') {
+				if (level == 0) {
+					Skript.error("Invalid placement of [optional brackets]");
+					return;
```

Comment:
We probably need to call `getParser().deleteCurrentEvent();` before each `return` statement